modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryNetworkIssuesTest.java

```diff
@@ -243,14 +253,31 @@ public void testBackwardNodeCheckWithSameLoopbackSingleLocalAddress() throws Exc
      */
     @Test
     public void testBackwardNodeCheckWithSameLoopbackSeveralLocalAddresses() throws Exception {
-        doTestBackwardNodeCheckWithSameLoopback("0.0.0.0");
```

Comment:
Let's move all this new test code to `doTestBackwardNodeCheckWithSameLoopback`. Except the "skipped" case, all the log messages are valid even if a node has only single address ('127.0.0.1').We would a siple `if` not to add `LogListener.matches(startLogMsg).andMatches("result=skipped")...`. We can determine it, for instance, by `spi(node0).getEffectiveNodeAddresses(node1.cluster().localNode())` before starting node3 and putting the log listeners to its configuration.